code/modules/mob/living/simple_animal/hostile/bees.dm

```diff
@@ -170,13 +170,14 @@
 
 	var/growth = health //Health also means how many bees are in the swarm, roughly.
 	//better healthier plants!
-	Hydro.health += round(growth*0.5)
+	Hydro.adjustHealth(growth*0.5)
 	if(prob(BEE_POLLINATE_PEST_CHANCE))
-		Hydro.pestlevel = max(0, --Hydro.pestlevel)
-	if(prob(BEE_POLLINATE_YIELD_CHANCE)) //Yield mod is HELLA powerful, but quite rare
-		Hydro.myseed.innate_yieldmod++
-	if(prob(BEE_POLLINATE_POTENTCY_CHANCE))
-		Hydro.myseed.potency++
+		Hydro.adjustPests(-10)
+	if(prob(BEE_POLLINATE_YIELD_CHANCE))
+		Hydro.myseed.adjust_yield(1)
+		Hydro.yieldmod = 2
```

Comment:
I assume this is supposed to be used for self-sustaining trays, correct? As this just gets reset to whatever `yieldmod` is calculated in the tray's `process()`.